sdk/dotnet/src/Microsoft.AspNetCore.Datasync/Tables/TableController.cs

```diff
@@ -298,7 +298,8 @@ public virtual async Task<IActionResult> QueryAsync(CancellationToken token = de
             Logger?.LogInformation("Query: {QueryString}", Request.QueryString);
             await AuthorizeRequest(TableOperation.Query, null, token).ConfigureAwait(false);
 
-            var dataset = Repository.AsQueryable()
+            var dataset = await Repository.AsQueryableAsync();
```

Comment:
Need to add a .ConfigureAwait(false) here (as we are a library)

Reply:
solved